src/parser/formatting-evaluator-module.ts

```diff
@@ -113,66 +114,83 @@ export class FormattingEvaluatorModule implements Module {
-    > = {};
+  _updateTagCount(
+    tagCount: Record<string, { regex: RegexCount; score: number; elementCount: number }>,
+    tagName: string,
+    regexCount: RegexCount,
+    score: number
+  ) {
+    // If we already had that tag included in the result, merge them and update total count
+    if (Object.keys(tagCount).includes(tagName)) {
+      for (const [k, v] of Object.entries(regexCount)) {
+        if (Object.keys(tagCount[tagName].regex).includes(k)) {
+          tagCount[tagName].regex[k] = {
+            ...tagCount[tagName].regex[k],
+            wordCount: tagCount[tagName].regex[k].wordCount + v.wordCount,
+          };
+          tagCount[tagName].elementCount += 1;
+        }
+      }
+    } else {
+      tagCount[tagName] = {
+        regex: regexCount,
+        score,
+        elementCount: 1,
+      };
+    }
+  }
+
+  _classifyTagsWithWordCount(htmlElement: HTMLElement, commentType: GithubCommentScore["type"]) {
+    const tagCount: Record<string, { regex: RegexCount; score: number; elementCount: number }> = {};
     const elements = htmlElement.getElementsByTagName("*");
 
     for (const element of elements) {
       const tagName = element.tagName.toLowerCase();
-      // We cannot use textContent otherwise we would duplicate counts, so instead we extract text nodes
-      const textNodes = Array.from(element?.childNodes || []).filter((node) => node.nodeType === 3);
-      const innerText = textNodes
-        .map((node) => node.nodeValue?.trim())
-        .join(" ")
-        .trim();
-      const symbols = this._countSymbols(this._multipliers[commentType].regex, innerText);
       let score = 0;
       if (this._multipliers[commentType]?.html[tagName] !== undefined) {
         score = this._multipliers[commentType].html[tagName];
-      } else {
-        logger.error(`Could not find multiplier for comment [${commentType}], <${tagName}>`);
-      }
-      logger.debug("Tag content results", { tagName, symbols, text: element.textContent });
-      // If we already had that tag included in the result, merge them and update total count
-      if (Object.keys(tagWordCount).includes(tagName)) {
-        for (const [k, v] of Object.entries(symbols)) {
-          if (Object.keys(tagWordCount[tagName].symbols).includes(k)) {
-            tagWordCount[tagName].symbols[k] = {
-              ...tagWordCount[tagName].symbols[k],
-              count: tagWordCount[tagName].symbols[k].count + v.count,
-            };
-          }
+        if (score === 0) {
+          element.remove();
+          continue;
         }
       } else {
-        tagWordCount[tagName] = {
-          symbols: symbols,
-          score,
-        };
+        logger.error(`Could not find multiplier for element <${tagName}> in comment [${element.outerHTML}]`);
+        element.remove();
+        continue;
       }
-    }
 
-    return tagWordCount;
+      // We cannot use textContent otherwise we would duplicate counts, so instead we extract text nodes
```

Comment:
Just use textContent from the top level parent element as we discussed several times. Copy my first implementation. The word counting was never complex.

Reply:
I noticed it doesn't always work after testing it in several issues, depending how it is formatted, which is why v1 would count the same words several times in different html elements, I can remove that fix if it doesn't matter.

Reply:
We also discussed that several times. The scoring, which is all that really matters, was correct regarding word count. The analytics counter wasn't.

Reply:
I was reading the spec again and this might be why I got confused:
```json
        "p": {
          "count": 7, // should be changed to `wordCount` and you should correctly count the amount of `p` tags to be `elementCount`
          "score": 1
        }
```

I thought you wanted to keep the count by element, otherwise if it is just the total words on the whole comment then `wordCount` should be at the root, not within elements themselves.

Reply:
Actually after checking the logic, using a global word count does not simplify the workflow because I would have to keep track for each count for each regex and then subtract the total at the end, which is also tedious. Currently the count is accurate the way it is and handles all the regexes, so let me know what you prefer.

Reply:
> I thought you wanted to keep the count by element, 

No. 

However we need to be able to exclude word count from `blockquote` and `code`.

Reply:
Gotcha, will move the word count to the top.